2k-joker/weaver-stock-tracker-app
# 3 implement stock lookup by company name

Fixes #3 

**Description:**
- Current implementation does not allow users to search stocks by `company name` - it only searches by `ticker symbol`.  My changes makes use of yahoo's finance api to solve this problem. See [this stack overflow post](https://stackoverflow.com/a/40243903/933972) for more details
- I also added `rubocop` gem to help standardize a coding style (style guide to come) 

**Dependencies:**
N/A

**Testing Instructions:**
- Run `rubocop` in terminal. All the tests should pass
- In rails console, run the following script and verify that you get a result from a valid search:
```
Stock.new_lookup('slack') 
 => #<Stock id: nil, ticker: "WORK", name: "Slack Technologies Inc", last_price: 0.431e2, created_at: nil, updated_at: nil, recent_performance: "-0.39%"> 
```
- Run the same script on `master` and verify that you get `nil` for the same search query:
```
2.6.6 :001 > Stock.new_lookup('slack')
 => nil 
```
- Note that 'slack' is just an example and you should try searching other companies as well
- Alternatively, you can test this on the UI by running the server locally, creating an account (if you don't already have one) and try searching for stocks on you **My Portfolio** page.
